### PR TITLE
Issue #45

### LINKED ISSUE #45: Review structs and messages classes inheritances
Review all structs and messages to make sure that is using proper class inheritance.

For instance, we have few Structs that are inheriting from GenericMessage, this is wrong.

Every struct that has a "header" attribute is a Message, otherwise is just a struct.

### PR DESCRIPTION
**Pull request by cemsbr**
Fix #45. I also made other changes as I was reading the code and put them in different commits.

### CODE REVIEW

File: pyof/v0x01/common/phy_port.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ class ListOfPhyPorts(basic_types.FixedTypeList):
     Attributes:
         items (optional): Instance or a list of instances of PhyPort
     """
-    def __init__(self, items=[]):
-        basic_types.FixedTypeList.__init__(self,
-                                           pyof_class=PhyPort,
-                                           items=items)
+    def __init__(self, items=None):
+        super().__init__(self,
```
Comment (beraldoleal) [inline pyof/v0x01/common/phy_port.py]: This is self on super() is deprecated on python3. Please fix.

Comment (cemsbr) [inline pyof/v0x01/common/phy_port.py]: As we can see in [PEP 3135](https://www.python.org/dev/peps/pep-3135/) and in [Python 3.5.1 docs](https://docs.python.org/3/library/functions.html#super), `super()` is not deprecated. Is this change still necessary?

Comment (diraol) [inline pyof/v0x01/common/phy_port.py]: Super is not deprecated. How is deprecated is the self argument you used on
the `super().__init__(self, ...)` Call.

Diego Rabatone Oliveira

Comment (cemsbr) [inline pyof/v0x01/common/phy_port.py]: Thanks for the clarification. Fixed and rebased.

Comment (beraldoleal) [inline pyof/v0x01/common/phy_port.py]: Yes, thanks @diraol . My line was not clear. Thanks @cemsbr .
